modules/dnn/src/layers/convolution_layer.cpp

```diff
@@ -438,6 +438,18 @@ class ConvolutionLayerImpl CV_FINAL : public BaseConvolutionLayerImpl
 #ifdef HAVE_OPENCL
         convolutionOp.release();
 #endif
+        if (inputs[0].dims == 3)
+        {
+            //Pool1D
```

Comment:
> modules/dnn/src/layers/convolution_layer.cpp
> Pool1D

Pool?

It is very unclear how condition correlates with the comment.
Looks like there are several assumptions which are just not mentioned here (they should be part of condition to have robust code)